### PR TITLE
Add EntityEvent.ChangeDimension

### PR DESCRIPTION
**Pull request by gegy1000**
I got it right this time I think.

### CODE REVIEW

File: src/main/java/xyz/openmodloader/event/impl/EntityEvent.java [hunk 1]
```diff
@@ -80,4 +81,73 @@ public boolean isCancelable() {
...
+         * @return the dimension this entity is travelling from.
+         */
+        public int getPreviousDimension() {
+            return previousDimension;
+        }
+
+        /**
+         * @return the dimension this entity is travelling to.
+         */
+        public int getNewDimension() {
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/EntityEvent.java]: rename to `setDimension`

File: src/main/java/xyz/openmodloader/event/impl/EntityEvent.java [hunk 2]
```diff
@@ -80,4 +81,73 @@ public boolean isCancelable() {
...
+            return newDimension;
+        }
+
+        /**
+         * Sets the dimension this entity will travel to.
+         */
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/EntityEvent.java]: JavaDocs are incomplete. You need to add parameters as well. The rest of the methods need body paragraphs as well. Look at your handle method for an example of a complete JavaDoc

**Comment by iLexiconn** [comment]
Looks good, fix merge conflics and I'll merge. :)

File: src/main/java/xyz/openmodloader/event/impl/EntityEvent.java [hunk 3]
```diff
@@ -80,4 +81,73 @@ public boolean isCancelable() {
...
+        /**
+         * Hook to make related patches much cleaner.
+         *
+         * @param entity the entity travelling through dimensions.
+         * @param previousDimension the dimension the entity is coming from.
+         * @param newDimension the dimension the entity is travelling to.
+         * @return the new dimension to travel to.
+         */
+        public static int handle(Entity entity, int previousDimension, int newDimension) {
+            EntityEvent.ChangeDimension event = new EntityEvent.ChangeDimension(entity, previousDimension, newDimension);
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/EntityEvent.java]: Event is never changed, so it should probably be marked as final.

**Comment by gegy1000** [comment]
Huh, it's missing a patch for EntityPlayerMP. Will fix.

File: src/main/java/xyz/openmodloader/event/impl/EntityEvent.java [hunk 4]
```diff
@@ -80,4 +81,74 @@ public boolean isCancelable() {
...
+        /**
+         * @return the dimension this entity is travelling to.
+         */
+        public int getNewDimension() {
+            return newDimension;
+        }
+
+        /**
+         * Sets the dimension this entity will travel to.
+         * @param newDimension the new dimension to travel to.
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/EntityEvent.java]: Close, but there should be a new line separating the description and the params :p

File: src/main/java/xyz/openmodloader/event/impl/EntityEvent.java [hunk 5]
```diff
@@ -80,4 +81,74 @@ public boolean isCancelable() {
...
+        }
+
+        /**
+         * @return the dimension this entity is travelling from.
+         */
+        public int getPreviousDimension() {
+            return previousDimension;
+        }
+
+        /**
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/EntityEvent.java]: Also these methods need a description body

**Comment by gegy1000** [comment]
@darkhax Is it really needed? The @return describes what the method is doing.

**Comment by gegy1000** [comment]
I'll do it anyway.

**Comment by gegy1000** [comment]
Okay, fixed.

**Comment by darkhax** [comment]
@gegy1000 Yes it is required. It's part of enforcing consistency in the code.

File: src/test/xyz/openmodloader/test/OMLTestMod.java [hunk 6]
```diff
@@ -81,6 +81,8 @@ public void onEnable() {
         OpenModLoader.INSTANCE.getEventBus().register(ArmorEvent.Equip.class, this::onArmorEquip);
         OpenModLoader.INSTANCE.getEventBus().register(ArmorEvent.Unequip.class, this::onArmorUnequip);
```
Comment (iLexiconn) [inline src/test/xyz/openmodloader/test/OMLTestMod.java]: Move the file back to the test dir, please

File: src/main/java/xyz/openmodloader/event/impl/EntityEvent.java [hunk 7]
```diff
@@ -80,4 +81,80 @@ public boolean isCancelable() {
...
+            return true;
+        }
+
+        /**
+         * Hook to make related patches much cleaner.
+         *
+         * @param entity the entity travelling through dimensions.
+         * @param previousDimension the dimension the entity is coming from.
+         * @param newDimension the dimension the entity is travelling to.
+         *
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/EntityEvent.java]: The new line shouldn't be below the params, only above. The point is to separate the body message.

**Pull request merged**